### DISCUSSION TITLE
semi auto properties with ref/in/out parameter

### DISCUSSION THREAD

**Discussion by zms9110750** [dotnet/csharplang#6083] [General]
part of my code:
```csharp
        TMP_Text text;
	public TMP_Text Text { get => text; protected set => text = value; }

	private void Awake()
	{ 
		Instantiate(path, transform).TryGetComponent(out text);
	}
```
In order to use the out parameter, I can only write one more field and change the automatic property to manual property.
Can the semi-automatic property  solve or consider solving this problem?

**Comment by huoyaoyuan** [comment]
Related: ref accessor (#1235)

**Comment by zms9110750** [comment]
In unity, attributes are not displayed in the edit panel, only fields are.
Sometimes I have to write another field for this reason.
I don't quite understand his serialization logic. However, if the fields in semi-automatic attributes can be exposed in some way, many problems may be solved?(this is a lazy way. If possible, I hope there is a better solution)

**Reply by CyrusNajmabadi** [reply]
@zms9110750 this sounds like something to take up with unity.

**Reply by FaustVX** [reply]
@zms9110750 
By Attributes, do you mean Properties ?
Attribute are really different than Properties.

And also, yes, it's a Unity issue, not a C# one.

**Reply by zms9110750** [reply]
Ah, yes, a translation problem. Such errors occur when our language uses ordinary translation software.
